src/transport/SecureSessionTable.cpp

```diff
@@ -99,32 +105,78 @@ SecureSession * SecureSessionTable::EvictAndAllocate(uint16_t localSessionId, Se
 
     auto cleanup = MakeDefer([this]() { mRunningEvictionLogic = false; });
 
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
+    const size_t kMaxSessionTableSize = mMaxSessionTableSize;
+#else
+    const size_t kMaxSessionTableSize = CHIP_CONFIG_SECURE_SESSION_POOL_SIZE;
+#endif
+
     ChipLogProgress(SecureChannel, "Evicting a slot for session with LSID: %d, type: %u", localSessionId,
                     (uint8_t) secureSessionType);
 
-    VerifyOrDie(mEntries.Allocated() <= CHIP_CONFIG_SECURE_SESSION_POOL_SIZE);
+    VerifyOrDie(mEntries.Allocated() <= kMaxSessionTableSize);
 
     //
     // Create a temporary list of objects each of which points to a session in the existing
     // session table, but are swappable. This allows them to then be used with a sorting algorithm
     // without affecting the sessions in the table itself.
     //
-    Platform::ScopedMemoryBufferWithSize<SortableSession> sortableSessions;
-    sortableSessions.Calloc(mEntries.Allocated());
-    if (!sortableSessions)
-    {
-        VerifyOrDieWithMsg(false, SecureChannel, "We couldn't allocate a session!");
-        return nullptr;
-    }
+    // The size of this shouldn't place significant demands on the stack - each item is
+    // 8 bytes in size (on a 32-bit platform), and 16 bytes in size (on a 64-bit platform,
+    // including padding). A minimal, spec compliant value for CHIP_CONFIG_SECURE_SESSION_POOL_SIZE
+    // is:
+    //
+    // (3 sessions / fabric * 5 fabrics + 1 reserved for CASE + 1 reserved for PASE) = 17 sessions.
+    //
+    // Total size of this stack variable = 17 * 8 = 136bytes (32-bit platform), 272 bytes (64-bit platform).
```

Comment:
That's the minimal size.  The actual size can be much larger, but I think the key is that this is under the control of the device definition.....  And that even if CHIP_CONFIG_SECURE_SESSION_POOL_SIZE is quite large (e.g. 1000) this is 16KB or so, which is not so bad on the sort of setup that would have that sort of pool size.

Reply:
Sure, I can add some text to that effect.